pkg/cache/context_recovery_test.go

```diff
@@ -106,83 +109,91 @@ func TestNodesRecovery(t *testing.T) {
 	dispatcher.Start()
 	defer dispatcher.Stop()
 
-	var node1 = v1.Node{
-		ObjectMeta: apis.ObjectMeta{
-			Name:      "host0001",
-			Namespace: "default",
-			UID:       "uid_0001",
-		},
-		Status: v1.NodeStatus{
-			Capacity: utils.NewK8sResourceList(
-				utils.K8sResource{
-					ResourceName: v1.ResourceMemory,
-					Value:        1024,
-				}, utils.K8sResource{
-					ResourceName: v1.ResourceCPU,
-					Value:        10,
-				}),
-		},
-	}
-
-	var node2 = v1.Node{
-		ObjectMeta: apis.ObjectMeta{
-			Name:      "host0002",
-			Namespace: "default",
-			UID:       "uid_0002",
-		},
-		Status: v1.NodeStatus{
-			Capacity: utils.NewK8sResourceList(
-				utils.K8sResource{
-					ResourceName: v1.ResourceMemory,
-					Value:        1024,
-				}, utils.K8sResource{
-					ResourceName: v1.ResourceCPU,
-					Value:        10,
-				}),
-		},
+	numNodes := 3
+	nodes := make([]*v1.Node, numNodes)
+	expectedStates := make([]string, numNodes)
+	for i := 0; i < numNodes; i++ {
+		nodes[i] = &v1.Node{
+			ObjectMeta: apis.ObjectMeta{
+				Name:      "host000%d" + strconv.Itoa(i),
+				Namespace: "default",
+				UID:       types.UID("uid_000" + strconv.Itoa(i)),
+			},
+			Status: v1.NodeStatus{
+				Capacity: utils.NewK8sResourceList(
+					utils.K8sResource{
+						ResourceName: v1.ResourceMemory,
+						Value:        1024,
+					}, utils.K8sResource{
+						ResourceName: v1.ResourceCPU,
+						Value:        10,
+					}),
+			},
+		}
+		expectedStates[i] = events.States().Node.Recovering
 	}
 
 	nodeLister := test.NewNodeListerMock()
-	nodeLister.AddNode(&node1)
-	nodeLister.AddNode(&node2)
+	for _, node := range nodes {
+		nodeLister.AddNode(node)
+	}
 	apiProvide4test.SetNodeLister(nodeLister)
 
 	mockedAppRecover := test.NewMockedRecoverableAppManager()
-	if err := context.recover([]interfaces.Recoverable{mockedAppRecover}, 3*time.Second); err == nil {
+	if err := context.recover([]interfaces.Recoverable{mockedAppRecover}, 1*time.Second); err == nil {
 		t.Fatalf("expecting timeout here!")
 	} else {
 		t.Logf("context stays waiting for recovery, error: %v", err)
 	}
```

Comment:
do we really need to know the content of the error if we are expecting it?
I know it was not introduced in this change but lets remove it

Reply:
Agree. it is fine to remove it.